repos/tap/components/list/treeList/treeList.js

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback } from 'react'
+import { pickKeys, noPropArr } from '@keg-hub/jsutils'
+import { useTheme } from '@keg-hub/re-theme'
+import { useSelector, shallowEqual } from 'react-redux'
+import { setFeatureActive } from 'SVActions/features'
+import { loadFeature } from 'SVActions/features/loadFeature'
+import {
+  View,
+  Loading,
+  Text,
+} from 'SVComponents'
+import { Values } from 'SVConstants'
+import TreeView from 'react-native-final-tree-view'
+import { ChevronDown } from 'SVAssets/icons'
+import { useStoreItems } from 'SVHooks/store/useStoreItems'
+
+const { CATEGORIES } = Values
+
+/**
+ * Finds the node based on given id
+ * @param {string} id 
+ * @param {Array<object>} nodes 
+ * 
+ * @returns {object} node
+ */
+const findNode = (id, nodes) => {
+  let foundNode = {}
+  nodes.reduce((__, node) => {
+    if (node.id === id) foundNode = node
+    if (node.children.length > 0) {
```

Comment:
No need to check if `length` is greater than `0`. Just do this => `node.children.length`
This could be cleaner in my opinion like this => 
```js
const findNode = (id, nodes) => {
  return nodes.reduce((foundNode, node) => {
    return foundNode.id
      ? foundNode
      : node.id === id
        ? node
        : node.children.length
          ? findNode(id, node.children)
          : foundNode
  }, {})
}
```
That code is untested, but I think you get the idea

Reply:
updated and cleaned up.
```
const findNode = (id, nodes) => 
  nodes.find((node) => {
    return node.id === id 
      ? node
      : node.children.length && findNode(id, node.children)
  }, {})
```

Reply:
Even better. Thanks